hnn_core/cell.py

```diff
@@ -4,12 +4,25 @@
 #          Sam Neymotin <samnemo@gmail.com>
 
 import numpy as np
+from numpy.linalg import norm
+
 from neuron import h, nrn
 
 # Units for e: mV
 # Units for gbar: S/cm^2
 
 
+def _get_yscale(p_secs, dipole_sec):
+    """Get cos theta to compute dipole in z direction."""
+    a = (np.array(p_secs[dipole_sec]['sec_pts'][1]) -
```

Comment:
Here we have the hard assumption that our cells live in the xy-plane. Not much to do about it, just a note.

Reply:
Not really. This just projects along the apical dendrite which may or may not be oriented along the y direction (Neuron coordinates ... z direction in physical coordinates). 

There are the following two alternatives which I considered and that does assume the xy-plane:

```py
y_long = (h.y3d(1, sec=sect) - h.y3d(0, sec=sect)) * pos_all
```

and 

```py
y_long = (p_secs[sec_name]['sec_pts'][1][1] - p_secs['apical_trunk']['sec_pts'][0][1]) * pos_all
```

and get rid of `yscale` at all. That didn't work though because of numerical issues I suspect. The code uses a mix of Neuron and Python. At some point we should figure out what kind of rounding/precision Neuron actually uses. We could deliberately typecast in Python then to make things match and then remove it when we're confident it's correct

Reply:
> Not really. This just projects along the apical dendrite which may or may not be oriented along the y direction

But isn't index `1` in `p_secs[dipole_sec]['sec_pts'][1]` explicitly the y-coordinate? And `0` on the next line x. That's what I mean by "hard assumptions"---we're doing 2D projection, implicitly ignoring the z-orientation, which is assumed to be degenerate (flat).

Reply:
Ah, my bad. All clear now!